src/rpc/orchestrator.ts

```diff
@@ -366,28 +364,16 @@ export async function deleteUser(_, { context: { user } }: RpcServerParams) {
   for (const workspace of workspaces) {
     await deleteAllAuthCredentials({}, { context: { user: { ...user, workspace: workspace.key, role: "admin" } } });
   }
-  const hubspotClient = new HubSpotClient({ accessToken: process.env.HS_PRIVATE_TOKEN });
-  const resp = await hubspotClient.crm.contacts.searchApi.doSearch({
-    filterGroups: [
-      {
-        filters: [
-          {
-            propertyName: "ceramic_did",
-            operator: "EQ",
-            value: userAccountId,
-          },
-        ],
-      },
-    ],
-    properties: ["email"],
-    limit: 100,
-    after: 0,
-    sorts: [],
-  });
-  deleteUserFromCache(userAccountId);
-  if (resp.results.length > 0) {
-    await hubspotClient.crm.contacts.batchApi.archive({ inputs: resp.results.map((x) => ({ id: x.id })) });
+  const usersCollection = await getCollection("users");
+  const userDoc = await usersCollection.findOne({ ceramic_did: userAccountId });
+  if (userDoc) {
+    const usersArchiveCollection = await getCollection("usersArchive");
+    const insertRes = await usersArchiveCollection.insertOne(userDoc);
```

Comment:
I would suggest to use [upsert](https://www.mongodb.com/docs/drivers/node/current/fundamentals/crud/write-operations/upsert/) here to avoid record duplication.

Reply:
@SAPikachu  Good idea, thanks. Also added "deletedAt" timestamp.